### DISCUSSION TITLE
ActiveMQ Connection Re-establishment After Retry Failure

### DISCUSSION THREAD

**Discussion by MehdiRifi** [Havret/dotnet-activemq-artemis-client#480] [Q&A] [answered]
Hello again,
It seems that connection do not automatically re-establish with ActiveMQ once the retry policy ends and the server comes back online. 
Is there a specific configuration I'm overlooking?

Thanks in advance!

**Comment by Havret** [comment]
Hello, 

Your issue might be due to the `RetryCount` setting for your recovery policy. If you want the connection to keep trying indefinitely once it drops and the server is back, avoid setting a `RetryCount`. This allows for endless recovery attempts. For a less formal setup, tweaking this setting might help. Check out the [documentation](https://havret.github.io/dotnet-activemq-artemis-client/docs/auto-recovery) for more on configuring recovery policies.

Best,
Havret

**Comment by MehdiRifi** [comment, marked as answer]
Hello @Havret, yes, I ended up not setting a RetryCount and implemented logic to cancel the message sending after a timeout. This prevents blocking the entire process, considering sending a message is an integral part of it.

Thanks a lot.
